src/transformers/modeling_utils.py

```diff
@@ -2209,14 +2209,40 @@ def gradient_checkpointing_enable(self, gradient_checkpointing_kwargs=None):
         if gradient_checkpointing_kwargs is None:
             gradient_checkpointing_kwargs = {"use_reentrant": True}
 
-        gradient_checkpointing_func = functools.partial(checkpoint, **gradient_checkpointing_kwargs)
+        if (
+            gradient_checkpointing_kwargs is not None 
+            and "use_deepspeed_grad_ckpt" in gradient_checkpointing_kwargs.keys()
+            and gradient_checkpointing_kwargs["use_deepspeed_grad_ckpt"]
+        ):
+            assert torch.distributed.is_initialized(), print("deepspeed ckpt need dist.initialization because it call dist.rank() for printing ...")
```

Comment:
```suggestion
            assert torch.distributed.is_initialized(), "deepspeed ckpt need dist.initialization because it call dist.rank() for printing ..."
```
This message is a bit confusing, what do you mean exactly by 

> deepspeed ckpt need dist.initialization because it call dist.rank() for printing ...

Reply:
without dist.init, i got error because [this line](https://github.com/microsoft/DeepSpeed/blob/3d347276ce80e1a29e777c839d1d7fabe8e5f034/deepspeed/runtime/activation_checkpointing/checkpointing.py#L540-L546) of deepspeed checkpoint class need dist.init to check rank